lib/chain.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+module.exports = function(api) {
+
+  const resolver = (resolve, reject) => (
+    (err, result) => {
+      if (err) reject(err);
+      else resolve(result);
+    }
+  );
+
+  class ArrayChain {
```

Comment:
`resolver` and `ArrayChain` can be moved out of this `module.exports =` function to reduce indentation levels. This would also indicate the fact that they are private to the module more clearly.